src/lib/builtins.pl

```diff
@@ -123,24 +205,38 @@
 
 % control operators.
 
+%% fail.
+%
+% A predicate that always fails
```

Comment:
The more declarative `false/0` (now part of the ISO standard) should be used instead.